src/Service/AppSearch/AppSearchService.php

```diff
@@ -82,6 +82,19 @@ public function addDocuments(array $items): BatchDocumentInterface
             }
 
             $fields = $this->getBuilder()->toArray($item);
+
+            $documentMaxSize = $this->getConfiguration()->getDocumentMaxSize();
+            if ($documentMaxSize  && strlen(serialize($fields)) >= $documentMaxSize) {
```

Comment:
Should we use `json_encode($fields)` here to make it more analogous to how Elastic will store it?

Reply:
Yeah good point. I'll change it.